autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClouderaManagerCommunicator.java

```diff
@@ -0,0 +1,72 @@
+package com.sequenceiq.periscope.monitor.handler;
+
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.api.swagger.HostsResourceApi;
+import com.cloudera.api.swagger.client.ApiClient;
+import com.sequenceiq.cloudbreak.client.HttpClientConfig;
+import com.sequenceiq.cloudbreak.cm.DataView;
+import com.sequenceiq.cloudbreak.cm.client.ClouderaManagerApiClientProvider;
+import com.sequenceiq.cloudbreak.cm.client.retry.ClouderaManagerApiFactory;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.ClusterManager;
+import com.sequenceiq.periscope.model.CloudInstanceType;
+import com.sequenceiq.periscope.service.ClusterService;
+import com.sequenceiq.periscope.service.security.TlsHttpClientConfigurationService;
+import com.sequenceiq.periscope.utils.ClusterUtils;
+
+@Service
+public class ClouderaManagerCommunicator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerCommunicator.class);
+
+    @Inject
+    private TlsHttpClientConfigurationService tlsHttpClientConfigurationService;
+
+    @Inject
+    private SecretService secretService;
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private ClouderaManagerApiClientProvider clouderaManagerApiClientProvider;
+
+    @Inject
+    private ClouderaManagerApiFactory clouderaManagerApiFactory;
+
+    @Cacheable(cacheNames = "cloudVMTypeCache", unless = "#result == null", key = "#cluster.id + #hostGroup")
+    public Optional<CloudInstanceType> getCloudVMDetailsForHostGroup(Cluster cluster, String hostGroup, Set<String> hostGroupFqdns) {
+        try {
+            LOGGER.debug("Retrieving CloudVMType for cluster '{}', hostGroup '{}'", cluster.getStackCrn(), hostGroupFqdns);
+            HttpClientConfig httpClientConfig = tlsHttpClientConfigurationService.buildTLSClientConfig(cluster.getStackCrn(),
+                    cluster.getClusterManager().getHost(), cluster.getTunnel());
+            ClusterManager cm = cluster.getClusterManager();
+            String user = secretService.get(cm.getUser());
+            String pass = secretService.get(cm.getPass());
+            ApiClient client = clouderaManagerApiClientProvider.getClient(Integer.valueOf(cm.getPort()), user, pass, httpClientConfig);
+            HostsResourceApi hostsResourceApi = clouderaManagerApiFactory.getHostsResourceApi(client);
```

Comment:
*Not really related to this patch*
Dug through the CM API after looking at this usage.
https://sseth-hf34092-master0.sseth-en.xcu2-8y8x.dev.cldr.work/api/v32/clusters/sseth-hf34092/services/yarn/roleConfigGroups/yarn-NODEMANAGER-WORKER/config - may provide the actual value allocated.
The problem here may be the roleConfigGroup name. Highly doubt that CB stores that in its database. Even if it does, or we could get access to the template to figure this out - determining the roleName may not be trivial.

Reply:
I tried hitting the compute roleConfigGroup but that does not contain the memory or vcpu https://10.112.6.147/api/v32/clusters/spot-distrox-5/services/yarn/roleConfigGroups/yarn-NODEMANAGER-COMPUTE/config

Response "items":[{"name":"process_auto_restart","value":"true","sensitive":false},{"name":"yarn_nodemanager_local_dirs","value":"/hadoopfs/fs1/nodemanager","sensitive":false},{"name":"yarn_nodemanager_log_dirs","value":"/hadoopfs/fs1/nodemanager/log","sensitive":false}]}"

Reply:
Sounds good. Which API did you end up using to look up the nodemanager memory. IThink you had mentioned a 'hosts' api.